### PR TITLE
Node 16

### PR DESCRIPTION
**Pull request by cwillisf**
### Description

Move to Node.js v16 and fix up `package-lock.json` (and `package.json` to a lesser extent). 

Note that `--use-legacy-peer-deps` is no longer necessary due to these changes -- hooray!

I re-added the step to install native dependencies to fix the build after the Node.js upgrade. Depending on the combination of Node.js version and npm dependency versions, sometimes these native dependencies are necessary and sometimes they're not. Installing them never hurts, other than a bit of extra time, so I think it's safer to leave them in. They're necessary when `node-gyp` fails to download a precompiled version of a library, probably `canvas`, which leads to a message like this: `npm ERR! node-pre-gyp WARN Pre-built binaries not found for canvas@2.6.1 and node@16.20.2 (node-v93 ABI, glibc) (falling back to source compile with node-gyp)`

### CODE REVIEW

File: .github/workflows/deploy.yml [hunk 1]
```diff
@@ -36,15 +36,14 @@ jobs:
             EOF
         - name: Setup
           run: |
-            # sudo apt update
-            # sudo apt install libgif-dev
+            sudo apt update
+            sudo apt install pkg-config libcairo2-dev libpango1.0-dev libssl-dev libjpeg62-dev libgif-dev
```
Comment (AlexisGoodfellow) [inline .github/workflows/deploy.yml:40]: I like the approach you've described with this, but (more for my own education than anything else 😅) how are we certain this is the complete set of dependencies which need to be installed? Just don't want to miss something!

File: .github/workflows/deploy.yml [hunk 2]
```diff
@@ -36,15 +36,14 @@ jobs:
...
             TRAVIS_BRANCH="${{github.ref_name}}" travis/setup-git.sh # Configure git user info
             travis/install-assets.sh
```
Comment (AlexisGoodfellow) [inline .github/workflows/deploy.yml:42]: Travis? 🤔

**Review by AlexisGoodfellow — APPROVED** [review]

File: .github/workflows/deploy.yml (hunk 2 above)
Comment (cwillisf) [reply .github/workflows/deploy.yml:42]: Ha! Yeah... upstream still uses Travis, so this should (in theory) make it easier to merge from them in the future.

File: .github/workflows/deploy.yml (hunk 1 above)
Comment (cwillisf) [reply .github/workflows/deploy.yml:40]: This is the list from [our README.md file](https://github.com/scratchfoundation/paper.js?tab=readme-ov-file#installing-native-dependencies-on-debianubuntu-linux), so if it's not sufficient then we should update both :)

(I'm pretty confident that it's sufficient)

**Pull request merged**